samples: logger: Fix test on stack hungry platforms

carlocaione opened the pull request:
One more CONFIG_TEST_EXTRA_STACKSIZE missing.

ioannisg reviewed (commented):
Looks OK to me. I leave this to @nordic-krch

carlocaione commented:
This is failing on shippable because forcing `LOG_PROCESS_THREAD_STACK_SIZE` to `4096` is too much on some platform. So I'll push this again leaving only the fix on `CONFIG_TEST_EXTRA_STACKSIZE` and defaulting `CONFIG_TEST_EXTRA_STACKSIZE` to `4096` only on the problematic platforms.

carlocaione committed:
samples: logger: Fix test on stack hungry platforms

This test is failing on platforms requiring a bigger stack size. This
issue is fixed by adding the missing CONFIG_TEST_EXTRA_STACKSIZE to
tweak the thread stack size.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

nordic-krch commented on samples/subsys/logging/logger/src/main.c:
```diff
@@ -30,7 +30,7 @@ static struct k_mem_partition *app_parts[] = {
 
 /* size of stack area used by each thread */
-#define STACKSIZE 1024
+#define STACKSIZE (1024 + CONFIG_TEST_EXTRA_STACKSIZE)
```
this looks a bit strange as define name suggests that it is test specific and yet it is used in a sample. Maybe stacksize should be bigger here?

Which platforms are more stack hungry? Maybe size should depend on whether it is 64 bit platform or not?

carlocaione replied:
it's one of the tests performed when using `sanitycheck` so it behaves like a test after all. I need a bigger stack size for the ARM64 port.

nordic-krch replied:
it's likely that 64 bit platform needs more stack thus i thought that may it should be something like
```
#define STACK_SIZE (1024 + ((IS_ENABLED(CONFIG_64BIT) ? 1024 : 0))
```
That is kind of a hint for a user that in his application he also need to consider different stack sizes for 64 bit platforms. I am not sure how fair is the assumption that 32 bit platforms use more or less same stack.

carlocaione replied:
While I agree with you I wonder what's the actual difference between tests and samples since both are actually tested during the sanity check? In the tests we have `CONFIG_TEST_EXTRA_STACKSIZE` that, even though it is misnamed for samples, it is exactly doing what we want and need.

I'm fine with both the solutions but I just think that we already have what we need, just with the wrong name maybe.

ioannisg replied:
@nordic-krch we typically leave this to the ARCHitecture, i.e. define the EXTRA Stack Size they need.

ioannisg replied:
We do similar things in other tests/samples
